apps/src/music/player/MusicLibrary.ts

```diff
@@ -96,16 +83,21 @@ export default class MusicLibrary {
     return this.folders.find(folder => folder.id === folderId) || null;
   }
 
-  // Given a folder path (e.g. "packs/pack1") return a SoundFolder.
-  getFolderForFolderPath(folderPath: string): SoundFolder | null {
-    return this.folders.find(folder => folder.path === folderPath) || null;
-  }
-
   // A progression step might specify a smaller set of allowed sounds.
   setAllowedSounds(allowedSounds: Sounds): void {
     this.allowedSounds = allowedSounds;
   }
 
+  generateSoundUrl(folder: SoundFolder, soundData: SoundData): string {
```

Comment:
could this just be `generateSoundUrlForId(id: string)` so you don't have to manually look up the `folder` and `soundData` first?

Reply:
If we have the ID (which is just `folder.id/sound.src`), we'd have to re-split it and look up the `SoundData` entry so that we could get the optional `path` which isn't included in the ID.